### PR TITLE
fix: public health/check should not include full details

### PR DESCRIPTION
**Pull request by wilr**
This PR limits the full disclosure of the checks to the dev/check and logged in users.

### CODE REVIEW

File: src/EnvironmentChecker.php
Comment (ChloeSartorelli) [inline src/EnvironmentChecker.php]: You have pretty much the same array being created 3 times here - can you please do this in a more DRY manner?

Comment (wilr) [reply src/EnvironmentChecker.php]: updated now

File: src/EnvironmentChecker.php
Comment (ChloeSartorelli) [inline src/EnvironmentChecker.php]: This should be determined based on the route (`health/check` vs `dev/check` i.e. are we using the `DevHealthController` or the `DevCheckController`?) - not on the type and validity of authentication.

`setIncludeDetails()` is public, so I'd recommend calling it from `DevCheckController::index()`: https://github.com/silverstripe/silverstripe-environmentcheck/blob/d3d20e5539acc01687d16ed5b53ad408eedfbfbe/src/Controllers/DevCheckController.php#L46-L48

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
I think this is appropriate.
The below message is still displayed along with the actual "Site is available" text
> you may check for the presence of the text 'Site is available' rather than an HTTP $ErrorCode error on this page, if you prefer

And according to the readme, this public URL is only meant to have some limited information.

Just one small thing to change though - plus if you wouldn't mind adding unit tests, that would be amazing.

**Comment by wilr** [comment]
Updated with unit test

File: tests/Controllers/DevHealthControllerTest.php
Comment (ChloeSartorelli) [inline tests/Controllers/DevHealthControllerTest.php:42]: We should probably also have one that checks `DevCheckController` _does_ include the details if that's okay.

File: src/Controllers/DevCheckController.php
Comment (ChloeSartorelli) [inline src/Controllers/DevCheckController.php]:
```suggestion
            ->setIncludeDetails(true);
```
This is the controller that _should_ include details. It's already false by default for the `DevHealthController`'s sake.

File: src/Controllers/DevCheckController.php
Comment (ChloeSartorelli) [inline src/Controllers/DevCheckController.php:51]: May as well string this onto the end with the other method calls since you've changed the calls above to be fluent

File: src/EnvironmentChecker.php
Comment (ChloeSartorelli) [inline src/EnvironmentChecker.php]:
```suggestion
```
Makes sense to add this for the `set*()` methods, but I don't think it makes sense for `log()`.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: src/Controllers/DevCheckController.php
Comment (wilr) [reply src/Controllers/DevCheckController.php:51]: `init` is not fluent though

Comment (ChloeSartorelli) [reply src/Controllers/DevCheckController.php:51]: If it's at the end of the chain that's fine - I'll leave this up to you though, I don't have strong feelings either way.

File: tests/Controllers/DevHealthControllerTest.php
Comment (wilr) [reply tests/Controllers/DevHealthControllerTest.php:42]: Done

File: src/EnvironmentChecker.php
Comment (michalkleiner) [inline src/EnvironmentChecker.php:271]: Can we also add at least a simple docblock?

Comment (wilr) [reply src/EnvironmentChecker.php:271]: Done

**Review by GuySartorelli — APPROVED** [review]
Looks good to me. Will give @michalkleiner a chance to comment if he wants to before merging.

**Pull request merged**
